api/src/test/java/org/pmiops/workbench/billing/FreeTierBillingServiceTest.java

```diff
@@ -329,27 +335,112 @@ public void checkFreeTierBillingUsage_workspaceMissingCreatorNoNPE() {
   }
 
   @Test
-  public void checkFreeTierBillingUsage_override() throws MessagingException {
+  public void maybeSetDollarLimitOverride_true() {
     workbenchConfig.billing.defaultFreeCreditsDollarLimit = 100.0;
-    doReturn(mockBQTableSingleResult(100.01)).when(bigQueryService).executeQuery(any());
+    final DbUser user = createUser(SINGLE_WORKSPACE_TEST_USER);
+    assertThat(user.getLastModifiedTime()).isNull();
+
+    // we update the user and should see this last modified time
+    final Instant time2 = START_INSTANT.plusSeconds(1000);
+    CLOCK.setInstant(time2);
+
+    assertThat(freeTierBillingService.maybeSetDollarLimitOverride(user, 200.0)).isTrue();
+    verify(mockUserServiceAuditor)
+        .fireSetFreeTierDollarLimitOverride(user.getUserId(), null, 200.0);
+    assertWithinBillingTolerance(freeTierBillingService.getUserFreeTierDollarLimit(user), 200.0);
+    assertThat(userDao.findUserByUserId(user.getUserId()).getLastModifiedTime())
+        .isEqualTo(new Timestamp(time2.toEpochMilli()));
+
+    // we update the user again and should see this new last modified time
+    final Instant time3 = START_INSTANT.plusSeconds(2000);
+    CLOCK.setInstant(time3);
+
+    assertThat(freeTierBillingService.maybeSetDollarLimitOverride(user, 100.0)).isTrue();
+    verify(mockUserServiceAuditor)
+        .fireSetFreeTierDollarLimitOverride(user.getUserId(), 200.0, 100.0);
+    assertWithinBillingTolerance(freeTierBillingService.getUserFreeTierDollarLimit(user), 100.0);
+    assertThat(userDao.findUserByUserId(user.getUserId()).getLastModifiedTime())
+        .isEqualTo(new Timestamp(time3.toEpochMilli()));
```

Comment:
I've been using `.isWithin(100.0).of(time_millis)` for millisecond comparisons. Have to cast the arg to `assertThat()` to `double` to make that work. I guess we could make an instant-valued version...